remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/api/ConfigApiClient.kt

```diff
@@ -47,20 +53,53 @@ internal class ConfigApiClient @VisibleForTesting constructor(
         throw InvalidRemoteConfigBaseUrlException(exception)
     }
 
-    fun fetchPath(path: String): Response {
-        val url = requestUrl.newBuilder()
-            .addPathSegments(path)
-            .build()
+    fun fetchPath(path: String, isConfig: Boolean): Response {
+        val builder = requestUrl.newBuilder().addPathSegments(path)
 
-        return client.newCall(
-            Request.Builder()
-                .url(url)
+        if (isConfig) {
+            try {
+                val pInfo: PackageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
+                builder.addQueryParameter(PARAM_KEY_APP_VERSION, pInfo.versionName)
+            } catch (e: PackageManager.NameNotFoundException) {
+                Log.d(SDK_NAME, e.localizedMessage.toString())
+            }
+
+            builder.addQueryParameter(PARAM_KEY_OS_VERSION, Build.VERSION.SDK_INT.toString())
+
+            val (language, country) = getLanguageDetails()
+            builder.addQueryParameter(PARAM_KEY_LANGUAGE, language)
+            if (country.isNotEmpty()) builder.addQueryParameter(PARAM_KEY_COUNTRY, country)
+        }
+
+        return client.newCall(Request.Builder()
+                .url(builder.build())
                 .build()
         ).execute()
     }
 
+    private fun getLanguageDetails(): Pair<String, String> {
+        val locale = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            context.resources.configuration.locales[0]
+        } else {
+            context.resources.configuration.locale
+        }
+        val lang = locale.language.toLowerCase(Locale.getDefault())
+
+        val country = try { locale.isO3Country.toLowerCase(Locale.getDefault())
```

Comment:
We had talked about the backend supporting both the 2 character or 3 digit version of the country code. So maybe this check isn't necessary? As long as the Locale is returning either of those.

Reply:
I understand. Changed to `Locale.country`